src/meta/src/rpc/service/notification_service.rs

```diff
@@ -95,7 +100,15 @@ where
             .collect_vec();
         let hummock_snapshot = Some(self.hummock_manager.get_last_epoch().unwrap());
 
-        let hummock_manager_guard = self.hummock_manager.get_read_guard().await;
```

Comment:
Would it be better to firstly pin version, then keep this read lock in line 98? So that

- we only hold read lock for latter slow ops.
- we may use slightly modified pin_version (remove last_pinned). Line 105 doesn't persist to meta store.

Reply:
We can remove pin_version and and from HummockMetaClient trait.

Reply:
> Would it be better to firstly pin version, then keep this read lock in line 98? So that
> 
> * we only hold read lock for following slow ops.
> * we may use slightly modified pin_version (remove last_pinned). Line 105 doesn't persist to meta store.

fixed

Reply:
> We can also remove pin_version and unpin_version from HummockMetaClient trait.

fixed